components/browser/state/src/main/java/mozilla/components/browser/state/reducer/BrowserStateReducer.kt

```diff
@@ -87,6 +91,36 @@ internal fun BrowserState.updateTabState(
     return this
 }
 
+/**
+ * Finds the corresponding tab in the [BrowserState] and replaces it using [update].
+ * @param tabId ID of the tab to change.
+ * @param update Returns a new version of [TabSessionState].
+ */
+@Suppress("Unchecked_Cast")
+internal fun BrowserState.updateTabState(
```

Comment:
I am not sure I want to have to update all reducers to these new methods. I might not fully understand why it's needed either. Let's pair later and see :).

Reply:
You convinced me. This is nicer :). Let's just add to the docs that this method does nothing for custom tabs!